core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/NamedViewResolver.java

```diff
@@ -0,0 +1,90 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.graph.hook;
+
+import uk.gov.gchq.gaffer.data.elementdefinition.view.NamedView;
+import uk.gov.gchq.gaffer.named.operation.cache.exception.CacheOperationFailedException;
+import uk.gov.gchq.gaffer.operation.Operation;
+import uk.gov.gchq.gaffer.operation.OperationChain;
+import uk.gov.gchq.gaffer.operation.Operations;
+import uk.gov.gchq.gaffer.operation.graph.OperationView;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.operation.handler.named.cache.NamedViewCache;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A {@link GraphHook} to resolve named operations.
+ */
+public class NamedViewResolver implements GraphHook {
+    private final NamedViewCache cache;
+
+    public NamedViewResolver() {
+        cache = new NamedViewCache();
+    }
+
+    public NamedViewResolver(final NamedViewCache cache) {
+        this.cache = cache;
+    }
+
+    @Override
+    public void preExecute(final OperationChain<?> opChain, final Context context) {
+        resolveViewsInOperations(opChain);
+    }
+
+    @Override
+    public <T> T postExecute(final T result, final OperationChain<?> opChain, final Context context) {
+        return null;
+    }
+
+    @Override
+    public <T> T onFailure(final T result, final OperationChain<?> opChain, final Context context, final Exception e) {
+        return null;
+    }
+
+    private void resolveViewsInOperations(final Operations<?> operations) {
+        final List<Operation> updatedOperations = new ArrayList<>(operations.getOperations().size());
+        for (final Operation operation : operations.getOperations()) {
+            if (operation instanceof OperationView) {
+                if (((OperationView) operation).getView() instanceof NamedView) {
+                    final NamedView resolvedNamedView = resolveNamedViewInOperation((NamedView) ((OperationView) operation).getView());
+
+                    final NamedView mergedNamedView = new NamedView.Builder().name(resolvedNamedView.getName()).merge(resolvedNamedView).merge(((OperationView) operation).getView()).build();
+                    ((OperationView) operation).setView(mergedNamedView);
+                }
+            }
+            updatedOperations.add(operation);
+        }
+        operations.getOperations().clear();
+        operations.getOperations().addAll((List) updatedOperations);
+    }
+
+    private NamedView resolveNamedViewInOperation(final NamedView opView) {
+        NamedView.Builder newNamedView;
+        try {
+            newNamedView = new NamedView.Builder().name(opView.getName()).merge(cache.getNamedView(opView.getName()));
+            for (final String name : opView.getMergedNamedViewNames()) {
+                newNamedView.merge(cache.getNamedView(name));
```

Comment:
Before you merge cache.getNamedView(name) in, you should recursively check it doesn't refer to any other named views.

Reply:
I don't think this is necessary.  If a NamedView is nested, then the parent will contain the NamedView name of the merged NamedViews, even if they are nested on several levels.  There is a test in to check.